### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by youmari**
## Added the following : 
- Install CanCanCan in your project.
- Add a role column to the users table. Remember to use migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin")
        - Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin").
        - Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -33,6 +34,18 @@ def create
...
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:id])
+    if @post.destroy
+      Post.update_user_posts_counter(@user)
+      flash[:success] = 'Post was successfully deleted.'
+    else
+      flash[:error] = 'Something went wrong'
+    end
+    redirect_to user_posts_url
+  end
```
Comment (dasileker) [inline app/controllers/posts_controller.rb]: - [ ] Nice work, but the destroy method is not working for the post when I try to delete, the update or delete on table "posts" violates foreign key constraint on table "likes"  is still referenced from table "likes".

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @youmari,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**

- No linter errors ✅.
- The PR is good ✅.
- The readme file is good ✅.


## Required Changes ♻️
_Check the comments under the review._
- as you can see you have double references in your tables post and likes that cause issues during deleting the post.


Capture web_5-4-2022_91346_localhost

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (youmari) [reply app/controllers/posts_controller.rb]: @dasileker that means you have to delete all the comments first then delete the post

**Review by dasileker — APPROVED** [review]
### **Hi @youmari,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**